src/state/slices/tradeInputSlice/selectors.ts

```diff
@@ -224,3 +223,93 @@ export const selectHasUserEnteredAmount = createSelector(
   selectInputSellAmountCryptoPrecision,
   sellAmountCryptoPrecision => bnOrZero(sellAmountCryptoPrecision).gt(0),
 )
+
+// All the below selectors are re-declared from tradeQuoteSlice/selectors to avoid circular deps
+// and allow selectSecondHopSellAccountId to keep a pwetty API
+
+const selectTradeQuoteSlice = (state: ReduxState) => state.tradeQuoteSlice
+const selectTradeQuotes = createDeepEqualOutputSelector(
+  selectTradeQuoteSlice,
+  tradeQuoteSlice => tradeQuoteSlice.tradeQuotes,
+)
+const selectSortedTradeQuotes = createDeepEqualOutputSelector(selectTradeQuotes, tradeQuotes => {
+  const allQuotes = Object.values(tradeQuotes)
+    .filter(isSome)
+    .map(swapperQuotes => Object.values(swapperQuotes))
+    .flat()
+  const happyQuotes = sortQuotes(allQuotes.filter(({ errors }) => errors.length === 0))
+  const errorQuotes = sortQuotes(allQuotes.filter(({ errors }) => errors.length > 0))
+  return [...happyQuotes, ...errorQuotes]
```

Comment:
gm, this works but we should prevent divergent logic with a central help for each case we're duplicating functionality. As an example I've implemented quote sorting in c987614aa353779f3cfb9830d46ad36203810f74

Reply:
Very valid! Indeed, those selectors duplications can bring in very sneaky... well, duplication of logic, which may be missed when refactoring one, but not the other. Went through all the duplicated selectors for core duplicated derivation logic that needs consolidation into helpers:

- `selectTradeQuoteSlice`: untouched, root slice selector
- `selectActiveQuoteMeta`: 8aa2975dec feat: getActiveQuoteMetaOrDefault. This consolidates the derivation logic in a shared `getActiveQuoteMetaOrDefault` helper along with some sane renaming of the selector from `selectActiveQuoteMeta` to `selectActiveQuoteMetaOrDefault`, and the previous `selectActiveQuoteMeta` is now semantically correct, selecting root `tradeQuoteSlice.activeQuoteMeta` without derivation. Also added some types for `ActiveQuoteMeta` while at it
- `selectSortedTradeQuotes` - tackled by yourself in https://github.com/shapeshift/web/commit/c987614aa353779f3cfb9830d46ad36203810f74 with a shared `sortTradeQuotes` @woodenfurniture 
- `selectActiveSwapperApiResponse` - didn't touch this guy as it doesn't do much and `tradeQuotes` is strongly typed on its first-level as `Partial<Record<SwapperName, Record<string, ApiQuote>>>`, so if there's ever a change in shape, we should spot it pretty easily regardless
- `selectConfirmedQuote` ditto untouched - root-level slice selector
- `selectActiveQuote`: Ditto untouched, as it simply returns active quote from swapper API response or from `tradeQuote.confirmedQuote)`
- `selectSecondHop`: Super simple guy, untouched (though the PR above brings in some nice safety re: steps accessooors)
- `selectIsActiveQuoteMultiHop`: Ditto untouched
- `selectSecondHopSellAccountId`: Untouched since it's only declared here and `tradeQuoteSlice/selectors` imports it